From 16504def8a6607948770919d401abee3a0909ff7 Mon Sep 17 00:00:00 2001
From: Aymeric PLOTON <aymeric.ploton@provenrun.com>
Date: Wed, 2 Aug 2023 19:54:52 +0200
Subject: [PATCH 5/5] Setting up and configuring 29-SPT and 14-REE interrupts
 for ATF

29: interruption of the secure physical timer for the secure world.
14: interrupt REE communication between the secure and non-secure worlds.
15: Interrupt 15 is reserved by the ATF, probably for core synchronization.
---
 plat/nxp/s32/include/s32_platform_def.h | 5 +++++
 plat/nxp/s32/s32_bl31.c                 | 7 ++++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/plat/nxp/s32/include/s32_platform_def.h b/plat/nxp/s32/include/s32_platform_def.h
index cb3b32457..867d6c124 100644
--- a/plat/nxp/s32/include/s32_platform_def.h
+++ b/plat/nxp/s32/include/s32_platform_def.h
@@ -8,6 +8,9 @@
 
 #include <common_def.h>
 
+/* Provencore SGI interrupt for REE */
+#define ARM_IRQ_SEC_SGI_6               14
+
 #define SIZE_1M                 (0x100000)
 
 #define S32_MPIDR_CPU_CLUSTER_MASK	0xFFF
@@ -225,5 +228,7 @@
 #define S32_SCP_CH_META_SIZE		(0)
 #endif
 
+#define BL32_IRQ_SEC_PHY_TIMER 29
+
 #endif /* S32_PLATFORM_H */
 
diff --git a/plat/nxp/s32/s32_bl31.c b/plat/nxp/s32/s32_bl31.c
index 936d2fa83..69177abc1 100644
--- a/plat/nxp/s32/s32_bl31.c
+++ b/plat/nxp/s32/s32_bl31.c
@@ -27,7 +27,8 @@
 #define MMU_ROUND_UP_TO_4K(x)	\
 	(((x) & ~0xfffU) == (x) ? (x) : ((x) & ~0xfffU) + 0x1000U)
 
-#define INTR_PROPS_NUM	2
+#define INTR_PROPS_NUM	4
+//#define INTR_PROPS_NUM	2
 #if defined(HSE_SUPPORT) && defined(SPD_opteed)
 #define MAX_INTR_PROPS	(INTR_PROPS_NUM + HSE_MU_INST)
 #else
@@ -160,6 +161,10 @@ static interrupt_prop_t interrupt_props[MAX_INTR_PROPS] = {
 		       INTR_GROUP0, GIC_INTR_CFG_EDGE),
 	INTR_PROP_DESC(S32CC_MSCM_CORE_0_IRQ, GIC_HIGHEST_SEC_PRIORITY,
 		       INTR_GROUP0, GIC_INTR_CFG_EDGE),
+	INTR_PROP_DESC(BL32_IRQ_SEC_PHY_TIMER, GIC_HIGHEST_SEC_PRIORITY,
+			INTR_GROUP1S, GIC_INTR_CFG_LEVEL),
+	INTR_PROP_DESC(ARM_IRQ_SEC_SGI_6, GIC_HIGHEST_SEC_PRIORITY,
+                        INTR_GROUP1S, GIC_INTR_CFG_LEVEL),
 };
 
 static unsigned int plat_s32_mpidr_to_core_pos(unsigned long mpidr);
-- 
2.34.1

